src/components/__tests__/TodoList.test.tsx

```diff
@@ -0,0 +1,28 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import TodoList from '../TodoList';
+
+describe('TodoList', () => {
+  it('should add a task to the list and store it to localStorage', () => {
+    render(<TodoList />);
+
+    fireEvent.change(screen.getByPlaceholderText('enter a task'), { target: { value: '123' } });
+
+    fireEvent.click(screen.getByRole('button'));
+
+    expect(screen.getByLabelText('list')).toBeInTheDocument();
+
+    const localTasks = JSON.parse(localStorage.getItem('localTasks') || '[]');
+    expect(localTasks[0].taskName).toEqual('123');
```

Comment:
Just repeating my comment from before. Rather then checking local storage just check if the item you added has been rendered to the screen.